File: Programmers/C++/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int res,ori,cnt;

int solution(int n) {
    ori = n;
    while(n) {
        cnt+=n%2;
        n/=2;
    }
    
    res=ori+1;
    while(res<1e6+1) {
        int k = res;
        int c = 0;
        while(k) {
            c+=k%2;
            k/=2;
        }
        if(cnt==c) return res;
        ++res;
    }
    return res;
}

File: Programmers/Swift/다음 큰 숫자.swift
import Foundation

var cnt = 0
var res = 0

func solution(_ N:Int) -> Int {
    var n = N
    while n>0 {
        cnt += n%2
        n/=2
    }
    
    var res = N+1
    while res<1000005 {
        var c = 0
        var k = res
        while k>0 {
            c += k%2
            k/=2
        }
        if c==cnt {return res}
        res+=1
    }
    
    return res
}
